### ISSUE TITLE
Change x-axis interval on Abacus plot

### ISSUE THREAD

**Issue by jdpye** [ocean-tracking-network/glatos#30]
In GitLab by @eschae on Nov 21, 2017, 13:32

I'd like to change the x-axis (Date/timestamp) of my abacus plot to weekly intervals, but haven't been able to find anything that works. My df is named "detections" and the code includes a control table and loop to spit out an abacus plot for each of my tagged individuals. 

abacusfun<-function(detections,transmitter_ids=50862){
  input<-subset(detections,detections$transmitter_id==transmitter_ids)
  plotTitle<-transmitter_ids
  name=i
  ext=".png"
  outfile=paste(name,ext)
  
  abacusPlot(input, detColNames=list(locationCol="receiver_sn", timestampCol="detection_timestamp_utcPOSIXct"),controlTable = controlTab, 
             outFile=outfile,plotTitle=plotTitle)
  
}

for (i in ids){
  
  graph=abacusfun(detections,transmitter_ids=i)

}

**Comment by jdpye** [comment]
In GitLab by @eschae on Nov 21, 2017, 13:33

changed the description

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Nov 21, 2017, 13:54

Unfortunately the x axis major ticks  in the abacusPlot function are hard-coded at five equidistant breaks over the range of x. The fastest way for you to get the spacing you want is probably to grab the source code (https://gitlab.oceantrack.org/GreatLakes/glatos/blob/master/R/abacusPlot.R) and edit lines 183-184 with something like: 

```{r}
 xmaj <- seq(from = min(detections$timestamp, na.rm = TRUE), 
       to = max(detections$timestamp, na.rm = TRUE), by = "week")
```

You'll then need to read in the function 'manually' (not called via the package), so you might also want to change the name of the function to distinguish this from the current glatos version.

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Nov 21, 2017, 13:59

In hindsight this function would probably serve best as a demo rather than a function because most users will need more flexibility than can realistically be accommodated in the function. An update to the glatos package is in the works that will focus more on interactive plots for data exploration and summarization and include code for plots like this as demos to be more easily accessed and modified locally by users.

**Comment by jdpye** [comment]
In GitLab by @eschae on Nov 21, 2017, 17:17

Thanks for the quick response! What do you mean by reading in the function 'manually'?

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Nov 21, 2017, 17:29

By 'read in the function manually' I just meant read in the function like any other function you write yourself (e.g., in this case since the abacusPlot source file only contains the one function, could hit Ctrl + Alt + R to 'run all' in RStudio after you've edited the file). 

The main point there was just that your customized function will be separate from the glatos package. Alternatively, you might just pick the relevant code out of the abacusPlot function and write your own to suit your needs.

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Feb 8, 2019, 14:53

requested functionality was added with a9ea93a9

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Feb 8, 2019, 14:58

mentioned in commit f42f4c220f82fd56a507c80233d1874926c64c19

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Feb 28, 2019, 08:45

fixed in 0.3.1

**Comment by jdpye** [comment]
In GitLab by @chrisholbrook on Feb 28, 2019, 08:45

closed
